src/runtime_src/core/pcie/driver/linux/xocl/subdev/cu.c

```diff
@@ -415,6 +416,18 @@ static int cu_probe(struct platform_device *pdev)
 
 	info = XOCL_GET_SUBDEV_PRIV(&pdev->dev);
 	BUG_ON(!info);
+
+	subdev_inst_idx = XOCL_SUBDEV_INST_IDX(&pdev->dev);
+	if (subdev_inst_idx == INVALID_INST_INDEX) {
+		XCU_ERR(xcu, "Unknown Instance index");
+		return -EINVAL;
+	}
+
+	/* Store subdevice instance index with this CU info.
+	 * This will be required to destroy this subdevice.
+	 */
+	info->inst_idx = subdev_inst_idx;
```

Comment:
@saifuddin-xilinx This change should also be made to scu.c, right?

Reply:
You are correct. Fixed. Thanks